kore/test/Test/Kore/Builtin/Set.hs

```diff
@@ -425,6 +428,35 @@ test_difference =
             (===) Pattern.top =<< evaluateT predicate
         )
 
+test_difference_symbolic :: TestTree
+test_difference_symbolic =
+    testCaseWithSMT
+        "SET.difference works for symbolic values"
+        (do
+            let patDifference =
+                    mkApplySymbol
+                        differenceSetSymbol
+                        [ xSingleton `concatSet` zeroSingleton
+                            `concatSet` oneSingleton
+                        , xSingleton `concatSet` zeroSingleton
+                        ]
+            actual <- evaluate patDifference
+            assertEqual "" expect actual
+        )
```

Comment:
This needs additional test cases.

If a symbolic key is found in the Set, we can remove it, but if it is not found, then we cannot conclude anything:

```
[X 1] -Set [X, Y] === [1] -Set [Y]
```

If a key might not be defined, then we need a `\ceil` condition:

```
[f(X) 1] -Set [f(X)] === [1] \and \ceil f(X)
```